spec/models/message_notifier_spec.rb

```diff
@@ -0,0 +1,204 @@
+require "spec_helper"
+
+describe MessageNotifier do
+  subject { MessageNotifier.new(user, ability, facility) }
+  let(:ability) { :stub_ability }
+  let(:facility) { order.facility }
+  let(:order) { create(:purchased_order, product: product) }
+  let(:order_detail) { order.order_details.first }
+  let(:product) { create(:instrument_requiring_approval) }
+  let(:user) { create(:user) }
+
+  let(:disputed_orders_visible?) { false }
+  let(:manage_training_requests?) { false }
+  let(:notifications_visible?) { false }
+  let(:show_problem_orders?) { false }
+  let(:show_problem_reservations?) { false }
+
+  before(:each) do
+    subject.stub(:disputed_orders_visible?).and_return(disputed_orders_visible?)
```

Comment:
Take a look at https://github.com/ryanb/cancan/wiki/Testing-Abilities#controller-testing

It might be nicer and fuller testing if you create a stub `Ability` object vs stubbing the private methods on the MessageNotifier. If you do that, I would almost move the `ability.can?`s into the `return [] unless @ability.can?(...)`. One less level of indirection.